src/plugins/robots/drone/simulator/pointmass3d_drone_model.cpp

```diff
@@ -116,13 +124,13 @@ namespace argos
       Real fTargetTransVelY =
          cPositionError.GetY() * XY_POS_KP;
       Real fTargetTransVelZ =
-         (m_cInputPosition.GetZ() - m_fTargetPositionZPrev) / GetPM3DEngine().GetPhysicsClockTick();
+         (m_cInputPosition.GetZ() - m_fPrevTargetAltitude) / GetPM3DEngine().GetPhysicsClockTick();
```

Comment:
The variable was called `m_fTargetPositionZPrev` because it makes this equation more readable